components/memory-cache/tests/memory-cache-component.spec.ts

```diff
@@ -0,0 +1,212 @@
+    })
+
+    it('should return matching keys for session pattern', async () => {
+      const result = await component.keys('session:*')
+
+      expect(result).toContain('session:abc')
+      expect(result).toContain('session:def')
+      expect(result).not.toContain('user:123')
+      expect(result).not.toContain('user:456')
+    })
+
+    it('should return empty array for non-matching pattern', async () => {
+      const result = await component.keys('nonexistent:*')
+      expect(result).toHaveLength(0)
+    })
+  })
+})
+
+describe('when handling different data types', () => {
+  describe('and storing string values', () => {
+    let value: string
+    
+    beforeEach(async () => {
+      value = 'test string'
+      await component.set('string-key', value)
+    })
+
+    it('should retrieve the string value', async () => {
+      const result = await component.get('string-key')
+      expect(result).toBe(value)
+    })
+  })
+
+  describe('and storing number values', () => {
+    let value: number
+    
+    beforeEach(async () => {
+      value = 42
+      await component.set('number-key', value)
+    })
+
+    it('should retrieve the number value', async () => {
+      const result = await component.get('number-key')
+      expect(result).toBe(value)
+    })
+  })
+
+  describe('and storing object values', () => {
+    let value: { id: number; name: string; active: boolean }
+    
+    beforeEach(async () => {
+      value = { id: 1, name: 'test', active: true }
+      await component.set('object-key', value)
+    })
+
+    it('should retrieve the object value', async () => {
+      const result = await component.get('object-key')
+      expect(result).toEqual(value)
+    })
+  })
+
+  describe('and storing array values', () => {
+    let value: (number | string | object)[]
+    
+    beforeEach(async () => {
+      value = [1, 'two', { three: 3 }]
+      await component.set('array-key', value)
+    })
+
+    it('should retrieve the array value', async () => {
+      const result = await component.get('array-key')
+      expect(result).toEqual(value)
+    })
+  })
+
+  describe('and storing null values', () => {
+    beforeEach(async () => {
+      await component.set('null-key', null)
+    })
+
+    it('should retrieve null', async () => {
+      const result = await component.get('null-key')
+      expect(result).toBeNull()
+    })
+  })
+})
+
+describe('when handling cache operations', () => {
+  it('should overwrite existing values', async () => {
+    const key = 'overwrite-key'
+    const firstValue = 'first'
+    const secondValue = 'second'
+
+    await component.set(key, firstValue)
+    let result = await component.get(key)
+    expect(result).toBe(firstValue)
+
+    await component.set(key, secondValue)
+    result = await component.get(key)
+    expect(result).toBe(secondValue)
+  })
+
+  it('should handle multiple concurrent operations without data corruption', async () => {
```

Comment:
As there shouldn't be any type of data corruption from executing an async set, WDYT about removing this test?